hs_core/models.py

```diff
@@ -2007,6 +2007,47 @@ def update(cls, element_id, **kwargs):
 
         super(FundingAgency, cls).update(element_id, **kwargs)
 
+    @classmethod
+    def migrate_from_crossref_ror(cls, apps, schema_editor):
+        """Migrate from CrossRef to ROR"""
+        logger = logging.getLogger(__name__)
+
+        for page in range(1, 500):
+            url = f"https://api.ror.org/v2/organizations?page={page}&filter=types:funder"
+            response = requests.get(url, verify=False)
+            if response.status_code == 200:
+                response_json = response.json()
+                logger.debug(f"Number of items: {response_json.get('number_of_results', None)}")
+                items = response_json.get('items', None)
+                if not items:
+                    break
+                logger.debug('Iterating through ROR items.')
+                for item in items:
```

Comment:
this migration will take a long time (5min?) EVEN IF WE ARE RUNNING IT ON A DB THAT HAS NO FUNDING AGENCIES! :)

Instead of getting all of the funding agencies from ROR and iterating through them, I wonder if it would be faster to iterate over all of the Funding Agencies that exist in our HydroShare db, and for each of those fundingAgencies, making a request to the ROR API...?

Let's see. We have ~5000 funding agencies in the HS db. Some of them have null agency_url
```py
>>> FundingAgency.objects.exclude(agency_url__isnull=True).count()
4558
```
Okay so maybe your approach is faster than making ~4558 synchronous requests. But we should consider making this asynchronous.

Here is an example task that we could use to ponder:
https://github.com/hydroshare/hydroshare/blob/2.15.8/hs_core/tasks.py#L434-L436

Reply:
After reviewing both options, I agree that many synchronous API calls would slow things down. I'll make it async and this speeds things up without blocking other tasks.